pkg/endpoint/endpoint_status_test.go

```diff
@@ -48,6 +49,15 @@ type endpointGeneratorSpec struct {
 	fakeControllerManager    bool
 }
 
+type endpointStatusConfiguration map[string]bool
```

Comment:
Would `map[string]struct{}` be sufficient?

Reply:
Technically, yes.  I think that the `bool` type self-documents the test better as it makes it obvious that a particular feature is enabled. Given these are unit tests, there is no need to micro-optimize memory use.